allennlp/common/params.py

```diff
@@ -354,6 +354,49 @@ def from_file(params_file: str, params_overrides: str = "") -> 'Params':
 
         return Params(param_dict)
 
+    def to_file(self, params_file: str, preference_orders: List[List[str]] = None):
+        with open(params_file, "w") as handle:
+            json.dump(self.as_ordered_dict(preference_orders), handle, indent=4)
+
+    def as_ordered_dict(self, preference_orders: List[List[str]] = None):
```

Comment:
Return type (`OrderedDict`).

Reply:
Actually dicts are by default ordered in python 3.6 onwards. link: https://stackoverflow.com/questions/39980323/are-dictionaries-ordered-in-python-3-6 . So skipping OrderedDict is good enough. I checked and tests also work.

Reply:
From that link: "This is considered an implementation detail in Python 3.6; you need to use OrderedDict if you want insertion ordering that's guaranteed across other implementations of Python. ... As of Python 3.7, this is no longer an implementation detail and instead becomes a language feature."

We're not on 3.7 yet.  Also, readers and writers of python code may not be aware that dictionaries are going to retain their insertion order, so it helps readability to be explicit about the ordering, especially when your function name says "ordered_dict".

Reply:
And you still need a type annotation either way =).

Reply:
Sure, I will use `OrderedDict` everywhere. Thanks

Reply:
done 👍